tests/integration/test_backup_with_other_granularity/test.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+
+
+from helpers.cluster import ClickHouseCluster
+cluster = ClickHouseCluster(__file__)
+
+
+node1 = cluster.add_instance('node1', with_zookeeper=True, image='yandex/clickhouse-server:19.4.5.35', stay_alive=True, with_installed_binary=True)
+node2 = cluster.add_instance('node2', with_zookeeper=True, image='yandex/clickhouse-server:19.4.5.35', stay_alive=True, with_installed_binary=True)
+node3 = cluster.add_instance('node3', with_zookeeper=True, image='yandex/clickhouse-server:19.4.5.35', stay_alive=True, with_installed_binary=True)
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_backup_from_old_version(started_cluster):
+    node1.query("CREATE TABLE source_table(A Int64, B String) Engine = MergeTree order by tuple()")
+
+    node1.query("INSERT INTO source_table VALUES(1, '1')")
+
+    assert node1.query("SELECT COUNT() FROM source_table") == "1\n"
+
+    node1.query("ALTER TABLE source_table ADD COLUMN Y String")
+
+    node1.query("ALTER TABLE source_table FREEZE PARTITION tuple();")
+
+    node1.restart_with_latest_version()
+
+    node1.query("CREATE TABLE dest_table (A Int64,  B String,  Y String) ENGINE = ReplicatedMergeTree('/test/dest_table1', '1')  ORDER BY tuple()")
+
+    node1.query("INSERT INTO dest_table VALUES(2, '2', 'Hello')")
+
+    assert node1.query("SELECT COUNT() FROM dest_table") == "1\n"
+
+    node1.exec_in_container(['bash', '-c', 'cp -r /var/lib/clickhouse/shadow/1/data/default/source_table/all_1_1_0/ /var/lib/clickhouse/data/default/dest_table/detached'])
+
+    assert node1.query("SELECT COUNT() FROM dest_table") == "1\n"
+
+    node1.query("ALTER TABLE dest_table ATTACH PARTITION tuple()")
```

Comment:
Do I understand correctly, that attaching of parts with non-adaptive granularity to a table with adaptive granularity is supported?

Reply:
Yes, it works. Each part checks it's granularity from FS https://github.com/clickhouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreeDataPartWide.cpp#L104 (method name is ugly).